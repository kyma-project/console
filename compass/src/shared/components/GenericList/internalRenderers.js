import React from 'react';
import { Popover } from 'fundamental-react/Popover';
import { Menu } from 'fundamental-react/Menu';
import { Button } from '@kyma-project/react-components';

export const renderActionElement = (actions, entry) =>
  !Array.isArray(actions) ? null : (
    <Popover
      body={
        <Menu>
          <Menu.List>
            {actions.map(action => (
              <Menu.Item
                onClick={() => action.handler(entry)}
                key={action.name}
              >
                {action.name}
              </Menu.Item>
            ))}
          </Menu.List>
        </Menu>
      }
      control={<Button glyph="vertical-grip" option="light" />}
      placement="bottom-end"
    />
  );
