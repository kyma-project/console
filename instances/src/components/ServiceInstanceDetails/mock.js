export const Mock = {
  __typename: 'clusterDocsTopic',
  name: 'kyma',
  groupName: 'root',
  displayName: 'Kyma',
  description: 'Overall documentation for Kyma',
  assets: [
    {
      __typename: 'clusterAsset',
      name: 'asset1',
      type: 'markdown',
      files: [
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/01-01-in-a-nutshell.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'In a nutshell',
            type: 'Overview',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-01-overview.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'Overview',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-03-cluster-installation.md',
          metadata: {
            __typename: 'fileMetadata',
            // title: 'Install Kyma on a cluster',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/12-01-try-out-kyma.md',
          metadata: {
            __typename: 'fileMetadata',
            type: 'Examples',
            title: 'Try out Kyma',
          },
        },
      ],
    },
  ],
};
