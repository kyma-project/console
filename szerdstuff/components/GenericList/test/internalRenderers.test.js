import { renderActionElement } from '../internalRenderers';
import { mockActions, mockEntry } from './mock';

describe('GenericList > internalRenderers', () => {
  describe('renderActionElement()', () => {
    it('Renders nothing if no args provided', async () => {
      expect(renderActionElement()).toMatchSnapshot();
    });

    it('Renders the popover if correct args are provided', async () => {
      expect(renderActionElement(mockActions, mockEntry)).toMatchSnapshot();
    });
  });
});
