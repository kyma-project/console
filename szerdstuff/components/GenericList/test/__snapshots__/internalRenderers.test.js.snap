// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericList > internalRenderers renderActionElement() Renders nothing if no args provided 1`] = `null`;

exports[`GenericList > internalRenderers renderActionElement() Renders the popover if correct args are provided 1`] = `
<Popover
  body={
    <Menu>
      <Menu.List>
        <Menu.Item
          onClick={[Function]}
        >
          action1
        </Menu.Item>
      </Menu.List>
    </Menu>
  }
  control={
    <Button
      glyph="vertical-grip"
      option="light"
    />
  }
  onClickOutside={[Function]}
  onEscapeKey={[Function]}
  placement="bottom-end"
/>
`;
